src/components/Snippet/FileManager.vue

```diff
@@ -0,0 +1,215 @@
+<script>
+import { setTimeout } from 'timers';
+
+import {
+  Upload as GirderUpload,
+  UpsertFolder as GirderUpsertFolder,
+  DataBrowser as GirderDataBrowser,
+  Breadcrumb as GirderBreadcrumb,
+} from '../';
+import {
+  getLocationType,
+  isRootLocation,
+  createLocationValidator,
+} from '../../utils';
+
+export default {
+  components: {
+    GirderBreadcrumb,
+    GirderUpload,
+    GirderUpsertFolder,
+    GirderDataBrowser,
+  },
+
+  props: {
+    location: {
+      type: Object,
+      validator: createLocationValidator(true),
+      default: () => { 'root'; },
+    },
+    rootLocationAllowed: {
+      type: Boolean,
+      default: false,
+    },
+    selectable: {
+      type: Boolean,
+      default: false,
+    },
+    dragEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    newFolderEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadMultiple: {
+      type: Boolean,
+      default: false,
+    },
+    uploadAccept: {
+      type: String,
+      default: '*',
+    },
+    preUpload: {
+      type: Function,
+      default: async () => {},
+    },
+    postUpload: {
+      type: Function,
+      default: async () => {},
+    },
+    preUpsert: {
+      type: Function,
+      default: async () => {},
+    },
+    postUpsert: {
+      type: Function,
+      default: async () => {},
+    },
+  },
+
+  inject: ['girderRest'],
+
+  data() {
+    return {
+      uploaderDialog: false,
+      newFolderDialog: false,
+      lazyLocation: null,
+    };
+  },
+
+  asyncComputed: {
+    async defaultLocation() {
+      const { user, apiRoot } = this.girderRest;
+      if (user) {
+        try {
+          return (await this.girderRest.get('folder', {
+            params: {
+              parentType: 'user',
+              parentId: user._id,
+            },
+          })).data[0];
+        } catch ({ response }) {
+          if (response) {
+            this.error = response.data.message;
+          } else {
+            this.error = `Could not connect to server: ${apiRoot}`;
+          }
+        }
+      } else {
+        return { type: 'root' };
+      }
+      return null;
+    },
+  },
+
+  computed: {
+    internalLocation: {
+      get() {
+        const { lazyLocation, girderRest } = this;
+        const { user } = girderRest;
+        if (lazyLocation) {
```

Comment:
This will cause this component has some non-prop-controllable behavior that is determined by some global state. So the logic should simply be if no location prop, root by default.

Reply:
Sure, if location is unspecified, default to root.